File: Cuarto/Programación Distribuida/java_concurrencia/src/Ejercicio2.java
public class Ejercicio2 {
    public static class MyThread extends Thread{
        int id;
        public MyThread(int id){
            this.id = id;
        }

        @Override
        public void run() {
            for (int i = 0; i < 5; i++) {
                System.out.println("ID: " + id);
                try {
                    sleep(1000);
                } catch (InterruptedException ignored) {}
                System.out.println("Hebra viva: " + this.isAlive());
            }
        }
    }
    public static void main(String[] args) {
        Thread t = new MyThread(1);
        t.start();
        try {
            t.join();
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        System.out.println("Hebra viva: " + t.isAlive());

        System.out.println("-------------------------------------------");

        Thread t1 = new MyThread(2);
        t1.start();
        try {
            t1.join(1000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        System.out.println("Hebra viva: " + t1.isAlive());
    }
}

File: Cuarto/ProgramaciónDistribuida/python_concurrencia/ejercicio2.py
import threading
import time

class myThread(threading.Thread):
    def __init__(self, id):
        threading.Thread.__init__(self)
        self.id = id
    
    def run(self):
        for _ in range(5):
            print("ID: ", self.id)
            time.sleep(1)
            print("La hebra sigue viva? ", self.is_alive())

thread1 = myThread(1)
thread1.start()
thread1.join(timeout=1)
if thread1.is_alive():
    print("La hebra sigue viva")